src/main/java/edu/dbsleipzig/stream/grouping/impl/algorithm/TableGroupingBase.java

```diff
@@ -614,34 +613,40 @@ protected Expression[] buildSelectGroupedVerticesGroupAttributes(){
     protected Expression buildJoinConditionForExpandedVertices(){
         PlannerExpressionSeqBuilder attributeJoinConditions = new PlannerExpressionSeqBuilder(getTableEnv());
         PlannerExpressionSeqBuilder temporalJoinConditions = new PlannerExpressionSeqBuilder(getTableEnv());
-        for (String key : vertexGroupingPropertyFieldNames.keySet()) {
-            attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key))
-              .isEqual($(vertexAfterGroupingPropertyFieldNames.get(key))));
-            attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key)).isNull()
-              .and($(vertexAfterGroupingPropertyFieldNames.get(key)).isNull()));
+        if (vertexGroupingPropertyKeys.size() > 0) {
```

Comment:
This really needed? If this is 0, then vertexGroupingPropertyFieldNames should be empty, too, or am I wrong?